### PR TITLE
spunk2argus

### PR DESCRIPTION
**Pull request by sbob909**
New example, allows for more flexible Splunk queries and assignment of
result set fields to Argus scope, metrics, and tags

### CODE REVIEW

File: examples/splunk2argus.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    # dictionary var to hold metrics
+    m_dict = {}
+    # for loop to populate m_dict
+    for row in data:
+
+        index = row["index"]
+
+        if not index:
+            logging.warn("Skipping row without index: %s", row)
+            continue
```
Comment (hdara-sfdc) [inline examples/splunk2argus.py]: Lines 170 to 174 seem to be leftover code? Rather, I guess you would want a generalized check inside the loop at line 181.

File: examples/splunk2argus.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+        # Every row has a timestamp
+        ts = row["_time"] and to_gmt_epoch(row["_time"])
+
+        # create final scope, substitute keys into the scope
+        rowScope = opts.argusscope
+        for keyName in keyNames:
+            rowScope = rowScope.replace("{"+keyName+"}",row[keyName])
```
Comment (hdara-sfdc) [inline examples/splunk2argus.py]: Check for existence of `keyName` in `row`.

File: examples/splunk2argus.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        # create tags
+        tag_dict = {}
+        for tagName in tagNames:
+            tag_dict[tagName] = row[tagName]
```
Comment (hdara-sfdc) [inline examples/splunk2argus.py]: Check for existence of `tagName` in `row`.

File: examples/splunk2argus.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+        # create metrics and datapoints
+        for col in metricNames:
+
+            m_key = (col)
+
+            val = row[col]
```
Comment (hdara-sfdc) [inline examples/splunk2argus.py]: Check for existence of `col` in `row`.

File: examples/splunk2argus.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            if "." in val:
+                val = float(val)
+            else:
+                val = int(val)
```
Comment (hdara-sfdc) [inline examples/splunk2argus.py]: If someone misconfigures tag columns as metric columns, it is possible to see non-numeric values here, so if we don't check for `val.isdigit()`, the end user would see an abrupt unhelpful exception.

**Comment by sbob909** [comment]
@hdara-sfdc thanks for the awesome feedback! I think I addressed all of your suggestions (and more) in the latest commit to my branch. Please review again (remember, I'm a python noob) and merge if there aren't any other findings.

**Pull request merged**